tests/tools/plugins/ssl_hook_test.cc

```diff
@@ -356,6 +378,7 @@ TSPluginInit(int argc, const char *argv[])
   }
 
   int preaccept_count       = 0;
+  int client_hello_count    = 0;
```

Comment:
You will want to make an immediate and a delayed version of the client hello hook test because it can be both reenabled immediately and reenabled later.  Should follow the same patter as for the cert callbacks.  Then in the test file put in cases for hello immediate and hello delayed.  Also exercise the case when there is more than one hello hook registered (count > 1).